openapi/src/main/java/io/micronaut/openapi/visitor/OpenApiExtraSchemaVisitor.java

```diff
@@ -0,0 +1,215 @@
+                    && isEmpty(packages)
+                    && isEmpty(excludePackages)) {
+                processExtraSchemaClass(element, context);
+                continue;
+            }
+            if (isNotEmpty(excludeClasses)) {
+                excludedExtraSchemaClassNames.addAll(Arrays.asList(excludeClasses));
+            }
+            if (isNotEmpty(excludeClassNames)) {
+                excludedExtraSchemaClassNames.addAll(Arrays.asList(excludeClassNames));
+            }
+            if (isNotEmpty(excludePackages)) {
+                excludedExtraSchemaPackages.addAll(Arrays.asList(excludePackages));
+            }
+            if (isNotEmpty(packages)) {
+                for (var packageName : packages) {
+                    var withSubpackages = false;
+                    if (packageName.endsWith(".*")) {
+                        packageName = packageName.substring(0, packageName.length() - 2);
+                        withSubpackages = true;
+                    }
+                    var classEls = getClassElements(packageName, context);
+                    for (var classEl : classEls) {
+                        processExtraSchemaClass(classEl, context);
+                    }
+                }
+            }
+            if (isNotEmpty(classes)) {
+                for (var className : classes) {
+                    var classEl = ContextUtils.getClassElement(className, context);
+                    processExtraSchemaClass(classEl, context);
+                }
+            }
+        }
+    }
+
+    private void processExtraSchemaClass(ClassElement classEl, VisitorContext context) {
+        if (classEl == null) {
+            return;
+        }
+        String schemaName = stringValue(classEl, io.swagger.v3.oas.annotations.media.Schema.class, PROP_NAME)
+                .orElse(computeDefaultSchemaName(null, classEl, classEl.getTypeArguments(), context, null));
+        var schema = getSchemaDefinition(resolveOpenApi(context), context, classEl, classEl.getTypeArguments(), null, Collections.emptyList(), null);
+        if (schema == null) {
+            return;
+        }
+        if (schema.get$ref().equals(COMPONENTS_SCHEMAS_PREFIX + schemaName)) {
+            var schemas = resolveSchemas(Utils.resolveOpenApi(context));
+            schema = schemas.get(schemaName);
+        }
+        extraSchemas.put(schemaName, schema);
+        extraSchemaClassnamesToNames.put(classEl.getName(), schemaName);
+    }
+
+    public static Map<String, Schema> getExtraSchemas() {
+        return extraSchemas;
+    }
+
+    public static Collection<String> getExcludedExtraSchemas() {
+        return excludedExtraSchemas;
+    }
+
+    public static void clean() {
+        extraSchemas = new HashMap<>();
+        extraSchemaClassnamesToNames = new HashMap<>();
+        excludedExtraSchemaClassNames = new HashSet<>();
+        excludedExtraSchemas = new HashSet<>();
+        excludedExtraSchemaPackages = new HashSet<>();
```

Comment:
these also have to be changed to `LinkedHashMap`

Reply:
fixed